### PR TITLE
fix: planning of consecutive fragments and fragments on union

### PR DESCRIPTION
**Pull request by devsergiy**
## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

**Review by jensneuse — APPROVED** [review]

File: router/core/graphql_prehandler.go [hunk 1]
```diff
@@ -614,6 +616,13 @@ func (h *PreHandler) handleOperation(req *http.Request, buf *bytes.Buffer, httpO
 		return nil, err
 	}
 
+	if h.queryPlansLoggingEnabled {
+		switch p := opContext.preparedPlan.preparedPlan.(type) {
+		case *plan.SynchronousResponsePlan:
+			h.log.Info(fmt.Sprintf("Query Plan:\n%s", p.Response.Fetches.QueryPlan().PrettyPrint()))
```
Comment (StarpTech) [inline router/core/graphql_prehandler.go:622]: Please use zap log fields

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router/core/graphql_prehandler.go (hunk 1 above)
Comment (StarpTech) [reply router/core/graphql_prehandler.go:622]: e.g.
```go
h.log.Info("Query Plan representation", zap.String("query_plan", p.Response.Fetches.QueryPlan().PrettyPrint()))
```

Comment (devsergiy) [reply router/core/graphql_prehandler.go:622]: it won't be readable in dev mode

anything structure getting printed escaped

Comment (devsergiy) [reply router/core/graphql_prehandler.go:622]: probably I could log with a field when dev mode disabled?

Comment (StarpTech) [reply router/core/graphql_prehandler.go:622]: Got it! I like your idea. Let's log it as a JSON field in production mode and in development as follow:

`s.logger.Sugar().Debugf("Query Plan:\n%s", ...)`

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**